### DISCUSSION TITLE
ublock origin is not a complete anti-capitalism replacement

### DISCUSSION THREAD

**Discussion by dark-veracity** [AprilSylph/XKit-Rewritten#256] [Suggestion Box]
The project board lists the anti-capitalism extension from the previous version of xkit as `will not implement` because the same functionality is provided more effectively by adblockers like ublock origin. Unfortunately adblocking on tumblr is only an incomplete solution. While the ads themselves are removed, empty sponsored post blocks are left behind littering the dashboard (see screenshot for an example). The anti-capitalism extension removed these for an overall better user experience and it would be good to see this feature return
<img width="577" alt="Screen Shot 2021-09-10 at 6 40 41 AM" src="https://user-images.githubusercontent.com/13202672/132848120-385f2339-f900-4ab8-a28f-46e3434bb86b.png">

**Comment by invalidCards** [comment]
It works fine if you enable more than just the default filters in uBO. I have every list enabled *except for* the following, and these empty sponsored blocks do not appear:
- Block access to LAN
- Fanboy's Annoyance
- Fanboy's Social
- Dan Pollock's hosts file
- MVPS Hosts
- (Anything under "Regions, languages")

**Comment by dark-veracity** [comment]
Oh interesting! After some experimentation it looks like the specific filter that removes the sponsored blocks is `uBlock filters – Annoyances`

That being said, I still think adding this to xkit rewritten is worth considering just to improve the user experience. Not everyone has the necessary expertise to poke around in their adblocker settings to turn this on (I didn't even realize there *were* non-default filters to enable until you pointed it out)

**Reply by AprilSylph** [reply]
That's kinda the point of not including it, though. Configuring uBlock Origin is something people should know how to do for their own sakes, and while people may not like being forced into a learning moment like this, I think it's more publicly valuable to do so than not.

**Reply by dark-veracity** [reply]
Hmm in that case maybe xkit rewritten should include some guidance on how to transition from the old anti-capitalism extension to ublock origin in its readme and in posts asking users to upgrade. What I want to avoid is the specific experience of a user trying the new addon, noticing that sponsored blocks are back, and thinking that this is just a missing feature and the new addon is inferior

**Comment by AprilSylph** [comment]


**Reply by dark-veracity** [reply]
That sounds like a good plan! Thank you for thinking about it
